starlite/handlers/websocket_message.py

```diff
@@ -0,0 +1,110 @@
+from inspect import Signature
+from typing import Any, Callable, Dict, List, Optional, Type, TypeVar, Union
+
+from pydantic import BaseModel, validate_arguments
+from pydantic.typing import AnyCallable
+
+from starlite.connection import WebSocket
+from starlite.exceptions import ImproperlyConfiguredException, ValidationException
+from starlite.provide import Provide
+from starlite.types import ExceptionHandler, Guard, Middleware
+from starlite.utils import is_async_callable
+
+from .base import BaseRouteHandler
+
+T_model = TypeVar("T_model", bound=BaseModel)
+
+
+class WSMessageHandler(BaseRouteHandler):
+    """
+    A websocket handler that wraps the handler function in a function that manages the open
+    websocket.
+
+    Handlers must specify the `data` kwarg with the type that messages should be coerced to.
+
+    Handlers may specify the `socket` kwarg, however any method on the socket that results
+    in `receive()` being called are not allowed to be called.
+
+    As many properties of the `HTTPRouteHandler` are supported, however they wrap the
+    socket connection, not each message.
+
+    A second set of properties that wrap the message are available.
+
+    Thinking
+    --------
+    - should I just use after_request and before_request as after_disconnect and before_accept?
```

Comment:
its a good idea to have lifecycle hooks for these.

Reply:
I'll leave considering all the hooks that should exist as a TODO for now.